src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -610,6 +610,15 @@ void Creature::Update(uint32 diff)
 
                     m_regenTimer += CREATURE_REGEN_INTERVAL;
                 }
+
+                if (CanNotReachTarget() && !IsInEvadeMode() && !GetMap()->IsRaid())
+                {
+                    m_cannotReachTimer += diff;
+                    if (m_cannotReachTimer >= 5000) // 5s?
+                        if (IsAIEnabled)
+                            AI()->EnterEvadeMode();
+                } 
+
```

Comment:
please move that `5000` to some constant, also the `if` could be rewritten as:

```cpp
if ( (m_cannotReachTimer >= SOME_CONST && IsAiEnabled)
{
   AI()->EnterEvadeMode();
}
```

Reply:
thanks for the catch, fixed